File: core/src/main/java/forpleuvoir/dhwu1a/core/event/group/GroupEventType.java
package forpleuvoir.dhwu1a.core.event.group;

/**
 * @author forpleuvoir
 * <p>#project_name dhwu1a
 * <p>#package forpleuvoir.dhwu1a.core.event.group
 * <p>#class_name GroupEventType
 * <p>#create_time 2021/7/3 21:45
 */
public enum GroupEventType {
    BotGroupPermissionChangeEvent(BotGroupPermissionChangeEvent.class),
    BotMuteEvent(BotMuteEvent.class),
    BotUnmuteEvent(BotUnmuteEvent.class),
    BotJoinGroupEvent(BotJoinGroupEvent.class),
    BotLeaveEventActive(BotLeaveEventActive.class),
    BotLeaveEventKick(BotLeaveEventKick.class),
    GroupRecallEvent(GroupRecallEvent.class),
    GroupNameChangeEvent(GroupNameChangeEvent.class),
    GroupEntranceAnnouncementChangeEvent(GroupEntranceAnnouncementChangeEvent.class),
    GroupMuteAllEvent(GroupMuteAllEvent.class),
    GroupAllowAnonymousChatEvent(GroupAllowAnonymousChatEvent.class),
    GroupAllowConfessTalkEvent(GroupAllowConfessTalkEvent.class),
    GroupAllowMemberInviteEvent(GroupAllowMemberInviteEvent.class),
    MemberJoinEvent(MemberJoinEvent.class),
    MemberLeaveEventKick(MemberLeaveEventKick.class),
    MemberLeaveEventQuit(MemberLeaveEventQuit.class),
    MemberCardChangeEvent(MemberCardChangeEvent.class),
    MemberSpecialTitleChangeEvent(MemberSpecialTitleChangeEvent.class),
    MemberPermissionChangeEvent(MemberPermissionChangeEvent.class),
    MemberMuteEvent(MemberMuteEvent.class),
    MemberUnmuteEvent(MemberUnmuteEvent.class),
    MemberHonorChangeEvent(MemberHonorChangeEvent.class);

    private final Class<? extends GroupEvent> clazz;

    GroupEventType(Class<? extends GroupEvent> clazz) {
        this.clazz = clazz;
    }

    public Class<? extends GroupEvent> getClazz() {
        return clazz;
    }

    public static boolean hasKey(String key) {
        var isHas = false;
        for (GroupEventType value : GroupEventType.values()) {
            isHas = value.equals(GroupEventType.valueOf(key));
            if (isHas) break;
        }
        return isHas;
    }
}

File: core/src/main/kotlin/forpleuvoir/dhwu1a/core/event/group/GroupEventType.kt
package forpleuvoir.dhwu1a.core.event.group

/**
 * @author forpleuvoir
 *
 * #project_name dhwu1a
 *
 * #package forpleuvoir.dhwu1a.core.event.group
 *
 * #class_name GroupEventType
 *
 * #create_time 2021/7/3 21:45
 */
enum class GroupEventType(val clazz: Class<out GroupEvent>) {
    BotGroupPermissionChangeEvent(forpleuvoir.dhwu1a.core.event.group.BotGroupPermissionChangeEvent::class.java),
    BotMuteEvent(forpleuvoir.dhwu1a.core.event.group.BotMuteEvent::class.java),
    BotUnmuteEvent(forpleuvoir.dhwu1a.core.event.group.BotUnmuteEvent::class.java),
    BotJoinGroupEvent(forpleuvoir.dhwu1a.core.event.group.BotJoinGroupEvent::class.java),
    BotLeaveEventActive(forpleuvoir.dhwu1a.core.event.group.BotLeaveEventActive::class.java),
    BotLeaveEventKick(forpleuvoir.dhwu1a.core.event.group.BotLeaveEventKick::class.java),
    GroupRecallEvent(forpleuvoir.dhwu1a.core.event.group.GroupRecallEvent::class.java),
    GroupNameChangeEvent(forpleuvoir.dhwu1a.core.event.group.GroupNameChangeEvent::class.java),
    GroupEntranceAnnouncementChangeEvent(forpleuvoir.dhwu1a.core.event.group.GroupEntranceAnnouncementChangeEvent::class.java),
    GroupMuteAllEvent(forpleuvoir.dhwu1a.core.event.group.GroupMuteAllEvent::class.java),
    GroupAllowAnonymousChatEvent(forpleuvoir.dhwu1a.core.event.group.GroupAllowAnonymousChatEvent::class.java),
    GroupAllowConfessTalkEvent(forpleuvoir.dhwu1a.core.event.group.GroupAllowConfessTalkEvent::class.java),
    GroupAllowMemberInviteEvent(forpleuvoir.dhwu1a.core.event.group.GroupAllowMemberInviteEvent::class.java),
    MemberJoinEvent(forpleuvoir.dhwu1a.core.event.group.MemberJoinEvent::class.java),
    MemberLeaveEventKick(forpleuvoir.dhwu1a.core.event.group.MemberLeaveEventKick::class.java),
    MemberLeaveEventQuit(forpleuvoir.dhwu1a.core.event.group.MemberLeaveEventQuit::class.java),
    MemberCardChangeEvent(forpleuvoir.dhwu1a.core.event.group.MemberCardChangeEvent::class.java),
    MemberSpecialTitleChangeEvent(forpleuvoir.dhwu1a.core.event.group.MemberSpecialTitleChangeEvent::class.java),
    MemberPermissionChangeEvent(forpleuvoir.dhwu1a.core.event.group.MemberPermissionChangeEvent::class.java),
    MemberMuteEvent(forpleuvoir.dhwu1a.core.event.group.MemberMuteEvent::class.java),
    MemberUnmuteEvent(forpleuvoir.dhwu1a.core.event.group.MemberUnmuteEvent::class.java),
    MemberHonorChangeEvent(forpleuvoir.dhwu1a.core.event.group.MemberHonorChangeEvent::class.java);

    companion object {
        fun hasKey(key: String): Boolean {
            var isHas = false
            for (value in values()) {
                isHas = value.name == key
                if (isHas) break
            }
            return isHas
        }
    }
}